mypy/test/teststubgen.py

```diff
@@ -108,17 +108,17 @@ def test_stubgen(testcase):
         sys.path.insert(0, 'stubgen-test-path')
     os.mkdir('stubgen-test-path')
     source = '\n'.join(testcase.input)
-    handle = tempfile.NamedTemporaryFile(prefix='prog_', suffix='.py', dir='stubgen-test-path',
-                                         delete=False)
+    with closing(tempfile.NamedTemporaryFile(prefix='prog_', suffix='.py',
+                                             dir='stubgen-test-path',
+                                             delete=False)) as handle:
+        handle.write(bytes(source, 'ascii'))
     assert os.path.isabs(handle.name)
```

Comment:
If anything goes wrong between this line and the `try` 6 lines below, the temp file will not be deleted. (In fact if the handle.write() call fails it also won't be deleted, as the close() alone doesn't delete it. If you really want to make this robust it probably requires extra checks on the cleanups in the `finally` clause to decide whether the cleanup should be done or not.